File: sdks/dotnet/src/Models/TargetingCondition.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace PhaseFlag.Models;

/// <summary>
/// A single condition within a targeting rule.
/// <para>
/// Supported operators: <c>is</c>, <c>is_not</c>, <c>contains</c>,
/// <c>not_contains</c>, <c>one_of</c>, <c>not_one_of</c>, <c>gt</c>,
/// <c>lt</c>, <c>matches_regex</c>.
/// </para>
/// </summary>
public sealed class TargetingCondition
{
    /// <summary>The context attribute to evaluate (e.g. "user_id", "country").</summary>
    [JsonPropertyName("attribute")]
    public string Attribute { get; set; } = string.Empty;

    /// <summary>The comparison operator.</summary>
    [JsonPropertyName("operator")]
    public string Operator { get; set; } = string.Empty;

    /// <summary>The target value to compare against.</summary>
    [JsonPropertyName("value")]
    public JsonElement Value { get; set; }
}

File: sdks/php/src/Models/TargetingCondition.php
<?php

declare(strict_types=1);

namespace PhaseFlag\Models;

/**
 * A single condition within a targeting rule.
 *
 * Conditions compare a named attribute from the {@see EvaluationContext}
 * against a target value using one of the supported operators:
 * is, is_not, contains, not_contains, one_of, not_one_of, gt, lt,
 * matches_regex.
 */
final class TargetingCondition
{
    public function __construct(
        public readonly string $attribute,
        public readonly string $operator,
        public readonly mixed $value,
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'attribute' => $this->attribute,
            'operator'  => $this->operator,
            'value'     => $this->value,
        ];
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): static
    {
        return new static(
            attribute: $data['attribute'] ?? '',
            operator: $data['operator'] ?? '',
            value: $data['value'] ?? null,
        );
    }
}
